src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.Executor.cs

```diff
@@ -23,39 +23,64 @@ namespace Microsoft.CodeAnalysis.Diagnostics.EngineV2
     internal partial class DiagnosticIncrementalAnalyzer
     {
         /// <summary>
-        /// Return all local diagnostics (syntax, semantic) that belong to given document for the given StateSet (analyzer) either from cache or by calculating them
+        /// Return all cached local diagnostics (syntax, semantic) that belong to given document for the given StateSet (analyzer).
```

Comment:
We split background diagnostic computation for open files into two steps:
1. Try to get cached diagnostics, while computing the set of analyzers that do not have cached diagnostics
2. Execute all the non-cached analyzers with a single invocation into CompilationWithAnalyzers.
3. Fetch computed diagnostics per-analyzer from the above invocation, and cache and raise diagnostic reported events

In near future, the diagnostic computation invocation into CompilationWithAnalyzers will be moved to OOP, which will help simplify and/or remove the IDE layer diagnostic caching in devenv process.

Reply:
It'd be good to include this explanation in the comments.

---
In reply to: [442895865](https://github.com/dotnet/roslyn/pull/45322#discussion_r442895865) [](ancestors = 442895865)